### PR TITLE
Remove head dependencies

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by orta** [comment]
:+1:

**Comment by mrackwitz** [comment]
:+1: LGTM

File: lib/cocoapods-core/dependency.rb [hunk 1]
```diff
@@ -109,12 +104,8 @@ def initialize(name = nil, *requirements)
...
 
       elsif requirements.last == :head
-        @head = true
-        requirements.pop
-        unless requirements.empty?
-          raise Informative, 'A `:head` dependency may not specify version ' \
-            "requirements (#{name})."
-        end
+        raise Informative, '`:head` dependencies have been removed. Please use ' \
+          "normal external source dependencies instead of `:head` for `#{name}`."
```
Comment (manuyavuz) [inline lib/cocoapods-core/dependency.rb]: Should we improve this message a little bit more? I couldn't clearly understand what to use instead when I read the message here. Maybe we should include an example like  `:git => 'REPO_PATH'` inside the message?

**Comment by manuyavuz** [comment]
:+1: other than the concern on informative message

File: lib/cocoapods-core/dependency.rb (hunk 1 above)
Comment (orta) [inline lib/cocoapods-core/dependency.rb]: That's a great idea @manuyavuz - is it easy to access the `git` repo of the dependency at this point? We could provide a copy & pastable version, if not, I think ^ is :+1:

Comment (segiddins) [inline lib/cocoapods-core/dependency.rb]: no, this is just a `Dependency` object, it has no idea what the latest `Specification` will be

File: lib/cocoapods-core/version.rb [hunk 2]
```diff
@@ -47,12 +47,9 @@ class Version < Pod::Vendor::Gem::Version
     #         A string representing a version, or another version.
     #
     def initialize(version)
-      if version.is_a?(Version) && version.head?
-        version = version.version
-        @head = true
-      elsif version.is_a?(String) && version =~ /HEAD based on (.*)/
+      if version.is_a?(String) && version =~ /HEAD based on (.*)/
```
Comment (segiddins) [inline lib/cocoapods-core/version.rb]: add `@todo` to remove this

**Pull request merged**